### ISSUE TITLE
On force-quit, project repos can be in bad state, ensure we reset --hard before pull

### ISSUE THREAD

**Issue by taquitos** [fastlane/ci#352]
eg:

```
[2018-03-22 12:03:47] DEBUG GitRepo:  Not setup yet, sleeping (time before timeout: 300.0) https://github.com/fastlane/ci
Thread terminated with exception: git '--git-dir=/Users/jliebowitz/.fastlane/ci/86286442-360f-4b9f-8e2b-d60cd0f257e7/.git' '--work-tree=/Users/jliebowitz/.fastlane/ci/86286442-360f-4b9f-8e2b-d60cd0f257e7' pull 'origin' 'master'  2>&1:From https://github.com/fastlane/ci
 * branch            master     -> FETCH_HEAD
error: The following untracked working tree files would be overwritten by merge:
	app/.angular-cli.json
	app/README.md
	app/karma.conf.js
	app/package.json
	app/src/app/app-routing.module.ts
	app/src/app/app.component.html
	app/src/app/app.component.ts
Please move or remove them before you can merge.
Aborting
  /Users/jliebowitz/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/git-1.3.0/lib/git/lib.rb:937:in `command'
  /Users/jliebowitz/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/git-1.3.0/lib/git/lib.rb:747:in `pull'
  /Users/jliebowitz/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/git-1.3.0/lib/git/base.rb:351:in `pull'
  /Users/jliebowitz/dev/ci/shared/models/git_repo.rb:303:in `block in pull'
  /Users/jliebowitz/dev/ci/shared/models/git_repo.rb:292:in `block in perform_block'
  /Users/jliebowitz/dev/ci/taskqueue/queue_worker.rb:82:in `block in process'
```

**Comment by minuscorp** [comment]
I tackled this by always cloning the repo if doesn't exists and resetting hard to its `sha` or `branch`, as it should've (because a project has always either a `branch` or a `sha` attached to it), to prevent this kind of issues.

**Comment by taquitos** [comment]
Yeah, the problem is that if the repo does exist, we still need to reset it 🙃

**Comment by minuscorp** [comment]
That's what I meant, I always checkout when branch or reset when sha. The repo is always in clean state regardless of the previous executions

**Comment by taquitos** [comment]
Should be fixed with #355
